src/widgets/upcoming-courses/ui/upcoming-courses.test.tsx

```diff
@@ -1,39 +1,58 @@
+/* eslint-disable import/no-namespace */
 import { render, screen } from '@testing-library/react';
 import { beforeEach, describe, expect, it, vi } from 'vitest';
 import { mockedCourses } from '@/shared/__tests__/constants';
+import * as helper from '@/shared/helpers/getActualData';
 import { UpcomingCourses } from '@/widgets/upcoming-courses';
 
-vi.mock('@/shared/helpers/getActualData', () => ({ getActualData: vi.fn().mockImplementation(() => mockedCourses) }));
+describe('UpcomingCourses', () => {
+  describe('with courses', () => {
+    beforeEach(async () => {
+      vi.spyOn(helper, 'getActualData').mockImplementation(() => mockedCourses);
+      const upcomingCourses = await UpcomingCourses();
 
-describe('Courses', () => {
-  beforeEach(async () => {
-    const upcomingCourses = await UpcomingCourses();
+      render(upcomingCourses);
+    });
 
-    render(upcomingCourses);
-  });
+    it('renders the title correctly', () => {
+      const titleElement = screen.getByText('Upcoming courses');
 
-  it('renders the title correctly', () => {
-    const titleElement = screen.getByText('Upcoming courses');
+      expect(titleElement).toBeInTheDocument();
+    });
 
-    expect(titleElement).toBeInTheDocument();
-  });
+    it('renders no more than 5 course cards', () => {
+      const coursesList = screen.getByTestId('courses-list');
+      const courseCardsWithoutButton = coursesList.children.length - 1;
 
-  it('renders no more than 5 course cards', () => {
-    const coursesList = screen.getByTestId('courses-list');
-    const courseCardsWithoutButton = coursesList.children.length - 1;
+      expect(courseCardsWithoutButton).toEqual(5);
+    });
 
-    expect(courseCardsWithoutButton).toEqual(5);
-  });
+    it('renders the Go to RS School button', () => {
+      const goButton = screen.getByText('Go to courses');
 
-  it('renders the Go to RS School button', () => {
-    const goButton = screen.getByText('Go to courses');
+      expect(goButton).toBeInTheDocument();
+    });
 
-    expect(goButton).toBeInTheDocument();
+    it('should render RsBanner', () => {
+      const rsBanner = screen.getByTestId('rs-banner');
+
+      expect(rsBanner).toBeInTheDocument();
+    });
   });
 
-  it('should render RsBanner', () => {
-    const rsBanner = screen.getByTestId('rs-banner');
+  describe('without courses', () => {
+    beforeEach(async () => {
+      mockedCourses.filter(() => []);
+      vi.spyOn(helper, 'getActualData').mockImplementation(() => []);
```

Comment:
line 45 seems redundant to me
I would use `vi.mocked` instead of `vi.spyOn`

Reply:
I return vi.mocked